sqlx-postgres/src/listener.rs

```diff
@@ -295,6 +295,19 @@ impl PgListener {
         }
     }
 
+    /// Receives the next notification that already exists in the connection buffer, if any.
+    ///
+    /// This is similar to `try_recv`, except it will not wait if the connection has not yet received a notification.
+    ///
+    /// This is helpful if you want to retrieve all buffered notifications and process them in batches.
+    pub fn try_recv_buffered(&mut self) -> Option<PgNotification> {
```

Comment:
The API is fine but the naming is a bit clunky. It's a shame we used `try_recv` for something else here. I think for that reason we should avoid `recv` for something that doesn't do I/O.

How about `.pop_buffered()`? Or `.next_buffered()`? I'd even be okay with `.pop()` though it may not be immediately obvious from context exactly what it does.

Reply:
I think `next_buffered()` is the most expressive of those options. But I also think that the presence/lack of `async` is a better indicator of I/O-ness than the word `recv`.

I also realized that `cargo test -p sqlx-postgres --all-features` isn't running the test I added, now I'm trying to get the integration tests running locally.

Reply:
Different naming makes it more obvious though. And it can be easy to miss whether a function is `async` or not, especially in autocomplete in IDEs, where often the only hint a function is `async` is whether it also recommends `.await` or not.